### PR TITLE
Refactor application of text decoration, text transform, letter spaci…

### PR DESCRIPTION
**Pull request by hamorphis**
…ng and formatted text.

Related to #2427

### CODE REVIEW

**Comment by hamorphis** [comment]
uitests

**Comment by hamorphis** [comment]
uitests

File: tns-core-modules/ui/text-field/text-field.ios.ts [hunk 1]
```diff
@@ -45,6 +45,10 @@ class UITextFieldDelegateImpl extends NSObject implements UITextFieldDelegate {
             }
 
             owner.dismissSoftInput();
+
+            if (owner.formattedText){
```
Comment (hshristov) [inline tns-core-modules/ui/text-field/text-field.ios.ts]: Is this needed only for iOS?

Comment (hamorphis) [inline tns-core-modules/ui/text-field/text-field.ios.ts]: Yes.

**Comment by hamorphis** [comment]
uitests

**Comment by hamorphis** [comment]
cuteness

**Comment by hamorphis** [comment]
widgets

**Pull request merged**